test/unit/test_anal_types.c

```diff
@@ -125,9 +205,120 @@ static bool test_anal_get_base_type_enum(void) {
 	mu_end;
 }
 
+static bool test_anal_save_base_type_enum(void) {
+	RAnal *anal = r_anal_new ();
+	mu_assert_notnull (anal, "Couldn't create new RAnal");
+	mu_assert_notnull (anal->sdb_types, "Couldn't create new RAnal.sdb_types");
+
+	RAnalBaseType *base = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_ENUM);
+	base->name = strdup ("foo");
+
+	RAnalEnumCase cas;
+	cas.name = strdup ("firstCase");
+	cas.val = 1;
+	r_vector_push (&base->enum_data.cases, &cas);
+
+	cas.name = strdup ("secondCase");
+	cas.val = 2;
+	r_vector_push (&base->enum_data.cases, &cas);
+
+	r_anal_save_base_type (anal, base);
+	r_anal_base_type_free (base);
+
+	Sdb *reg = sdb_new0 ();
+	setup_sdb_for_enum (reg);
+	assert_sdb_eq (anal->sdb_types, reg, "save enum type");
+	sdb_free (reg);
+
+	r_anal_free (anal);
+	mu_end;
+}
+
+static bool test_anal_get_base_type_typedef(void) {
+	RAnal *anal = r_anal_new ();
+	mu_assert_notnull (anal, "Couldn't create new RAnal");
+	mu_assert_notnull (anal->sdb_types, "Couldn't create new RAnal.sdb_types");
+
+	setup_sdb_for_typedef (anal->sdb_types);
+
+	RAnalBaseType *base = r_anal_get_base_type (anal, "string");
+	mu_assert_notnull (base, "Couldn't create get base type of typedef \"string\"");
+
+	mu_assert_eq (R_ANAL_BASE_TYPE_KIND_TYPEDEF, base->kind, "Wrong base type");
+	mu_assert_streq (base->name, "string", "type name");
+	mu_assert_streq (base->type, "char *", "typedefd type");
+
+	r_anal_base_type_free (base);
+	r_anal_free (anal);
+	mu_end;
+}
+
+static bool test_anal_save_base_type_typedef(void) {
+	RAnal *anal = r_anal_new ();
+	mu_assert_notnull (anal, "Couldn't create new RAnal");
+	mu_assert_notnull (anal->sdb_types, "Couldn't create new RAnal.sdb_types");
+
+	RAnalBaseType *base = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_TYPEDEF);
+	base->name = strdup ("string");
+	base->type = strdup ("char *");
+
+	r_anal_save_base_type (anal, base);
+	r_anal_base_type_free (base);
+
+	Sdb *reg = sdb_new0 ();
+	setup_sdb_for_typedef (reg);
+	assert_sdb_eq (anal->sdb_types, reg, "save typedef type");
+	sdb_free (reg);
+
+	r_anal_free (anal);
+	mu_end;
+}
+
+static bool test_anal_get_base_type_atomic(void) {
+	RAnal *anal = r_anal_new ();
+	mu_assert_notnull (anal, "Couldn't create new RAnal");
+	mu_assert_notnull (anal->sdb_types, "Couldn't create new RAnal.sdb_types");
+
+	setup_sdb_for_atomic (anal->sdb_types);
+
+	RAnalBaseType *base = r_anal_get_base_type (anal, "char");
+	mu_assert_notnull (base, "Couldn't create get base type of atomic type \"char\"");
+
+	mu_assert_eq (R_ANAL_BASE_TYPE_KIND_ATOMIC, base->kind, "Wrong base type");
+	mu_assert_streq (base->name, "char", "type name");
+	mu_assert_streq (base->type, "c", "atomic type type");
```

Comment:
sorry for the stupid question... `c` stands for what? And the `base->size` is in bits?

Reply:
`c` is char, which comes from the sdb format. Yes, `base->size` is in bits, there is a comment about it in the struct definition.

Reply:
c is char in pf format, not sdb

Reply:
It is also used in sdb, for example here:
https://github.com/radareorg/radare2/blob/fe5b80d615881bd1c31e0a99d655f4af90acf126/libr/anal/d/types.sdb.txt#L41
In the long run, this should probably be made an enum (and types should be outside of sdb) but that is the current format.

Reply:
yep i was just trying to clarify

Reply:
Ok, now I understand what you actually meant by that. I think we shouldn't save the concrete pf format in the sdb but just something like "this is some char", "this is some float", "this is signed/unsigned" from which in combination with the bitsize the pf format could be easily derived. I think I'll open an issue for that.

Reply:
https://github.com/radareorg/radare2/issues/17649